src/officehours_api/backends/zoom.py

```diff
@@ -152,44 +134,48 @@ def _get_access_token(cls, user: User) -> str:
         return user.profile.backend_metadata['zoom']['access_token']
 
     @classmethod
-    def _get_session(cls, user: User) -> requests.Session:
-        session = requests.Session()
-        session.headers.update({
-            'Authorization': f'Bearer {cls._get_access_token(user)}',
-            'Accept': 'application/json',
-            'Content-Type': 'application/json',
-        })
-        return session
+    def _get_client(cls, user: User) -> ZoomClient:
+        """Gets a ZoomClient instance for the given user. Replaces the _get_session method."""
+        zoom_meta = user.profile.backend_metadata['zoom']
+        return ZoomClient(zoom_meta['access_token'])
 
     @classmethod
     def _create_meeting(cls, user: User) -> ZoomMeeting:
-        session = cls._get_session(user)
+        """Creates a Zoom meeting for the given user."""
+        client = cls._get_client(user)
         user_id = user.profile.backend_metadata['zoom']['user_id']
-        resp = session.post(
-            f'{cls.base_url}/v2/users/{user_id}/meetings',
-            json={
-                'topic': 'Remote Office Hours Queue Meeting',
-                'start_time': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ'),
-                'timezone': 'America/Detroit',
-                "settings": {
-                    "join_before_host": False,
-                    "waiting_room": True,
-                    "meeting_authentication": False,
-                    "use_pmi": False
-                },
-            },
+        # Specify the meeting settings
+        # We are initializing the meeting settings with the default settings
+        # and then modifying the settings we want to change
+        meeting_settings = ZoomMeetingSettings.default_settings()
+        meeting_settings.waiting_room = True
+        meeting_settings.join_before_host = False
+        meeting_settings.meeting_authentication = False
+        meeting_settings.use_pmi = False
+        meeting_settings.contact_name = user.first_name + ' ' + user.last_name
+        meeting_settings.contact_email = user.email
+        # invoke the create_meeting method of the ZoomClient instance
+        meeting = client.meetings.create_meeting(
+            topic='Remote Office Hours Queue Meeting',
+            start_time=datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ'),
+            duration_min=60,
+            timezone='America/Detroit',
+            settings=meeting_settings
         )
-        if resp.status_code == 401:
-            logger.info(f'Access token for user {user.id} seems to be invalid.')
-            cls._clear_backend_metadata(user)
-        resp.raise_for_status()
-        logger.debug("Created meeting: %s", resp.json())
-        return resp.json()
+        # The .json() method does not seem to work all that well
```

Comment:
What do you mean by "The .json() method does not seem to work all that well"?

Reply:
The return value of `meeting.json()` is a string object instead of a Python dictionary object. Thus necessities the use of the `json.loads()` function.

Reply:
Fixes in commit peijli/remote-office-hours-queue@77ef326